### PR TITLE
Http4s syntax update

### PR DESCRIPTION
**Pull request by janstenpickle**
Addresses #43 

Uses a similar method to https://github.com/http4s/http4s/pull/3552 for working with a Kliesli injected span.

Also adds functionality to `Span` whereby a `PartialFunction[Throwable, SpanStatus]` may be used to capture specific errors and map them to a span status. This is specifically used in Http4s `ClientSyntax` where if the body is read and an `UnexpectedStatus` exception is thrown, it may be handled properly.

### CODE REVIEW

File: modules/newrelic-http-exporter/src/main/scala/io/janstenpickle/trace4cats/newrelic/NewRelicSpanExporter.scala [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
 import org.http4s.client.Client
-import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.client.blaze.BlazeClientBuilder
 import org.http4s.headers.`Content-Type`
 import org.http4s.{Header, MediaType}
 
 object NewRelicSpanExporter {
 
-  def emberClient[F[_]: Concurrent: Timer: ContextShift: Logger](
+  def emberClient[F[_]: ConcurrentEffect: Timer: ContextShift](
```
Comment (catostrophe) [inline modules/newrelic-http-exporter/src/main/scala/io/janstenpickle/trace4cats/newrelic/NewRelicSpanExporter.scala:16]: ember -> blaze

File: modules/newrelic-http-exporter/src/main/scala/io/janstenpickle/trace4cats/newrelic/NewRelicSpanCompleter.scala [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 import io.janstenpickle.trace4cats.model.TraceProcess
 import org.http4s.client.Client
-import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.client.blaze.BlazeClientBuilder
 
 import scala.concurrent.duration._
 
 object NewRelicSpanCompleter {
-  def emberClient[F[_]: Concurrent: Timer: ContextShift: Logger](
+  def emberClient[F[_]: ConcurrentEffect: Timer: ContextShift](
```
Comment (catostrophe) [inline modules/newrelic-http-exporter/src/main/scala/io/janstenpickle/trace4cats/newrelic/NewRelicSpanCompleter.scala:15]: ember -> blaze

File: modules/datadog-http-exporter/src/main/scala/io/janstenpickle/trace4cats/datadog/DataDogSpanExporter.scala [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
 import io.janstenpickle.trace4cats.model.Batch
 import org.http4s.Method.PUT
 import org.http4s.circe.CirceEntityCodec._
 import org.http4s.client.Client
-import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.client.blaze.BlazeClientBuilder
 
 object DataDogSpanExporter {
-  def emberClient[F[_]: Concurrent: Timer: ContextShift: Logger](
+  def emberClient[F[_]: ConcurrentEffect: Timer: ContextShift: Logger](
```
Comment (catostrophe) [inline modules/datadog-http-exporter/src/main/scala/io/janstenpickle/trace4cats/datadog/DataDogSpanExporter.scala:14]: ember -> blaze

File: modules/datadog-http-exporter/src/main/scala/io/janstenpickle/trace4cats/datadog/DataDogSpanCompleter.scala [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 import io.janstenpickle.trace4cats.model.TraceProcess
 import org.http4s.client.Client
-import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.client.blaze.BlazeClientBuilder
 
 import scala.concurrent.duration._
 
 object DataDogSpanCompleter {
-  def emberClient[F[_]: Concurrent: Timer: ContextShift: Logger](
+  def emberClient[F[_]: ConcurrentEffect: Timer: ContextShift](
```
Comment (catostrophe) [inline modules/datadog-http-exporter/src/main/scala/io/janstenpickle/trace4cats/datadog/DataDogSpanCompleter.scala:15]: ember -> blaze

**Review by catostrophe — COMMENTED** [review]
Need to rename some methods

**Comment by janstenpickle** [comment]
Thanks @catostrophe!

**Pull request merged**